Libplanet/Net/Swarm.cs

```diff
@@ -2163,6 +2025,116 @@ private void TransferRecentStates(GetRecentStates getRecentStates)
             Transport.ReplyMessage(reply);
         }
 
+        private async Task ProcessFillblock(CancellationToken cancellationToken)
+        {
+            while (!cancellationToken.IsCancellationRequested)
+            {
+                if (_demandBlockHash is null ||
+                    _demandBlockHash.Value.Item1 <= BlockChain.Tip.Index)
+                {
+                    await Task.Delay(100, cancellationToken);
+                }
+                else
+                {
+                    (long index, BoundPeer peer, HashDigest<SHA256> blockHash) =
+                        _demandBlockHash.Value;
+                    if (index > BlockChain.Tip.Index)
+                    {
+                        try
+                        {
+                            await SyncPreviousBlocksAsync(
+                                BlockChain,
+                                peer,
+                                blockHash,
+                                null,
+                                0,
+                                true,
+                                cancellationToken);
+                            BlockReceived.Set();
+                            BlockAppended.Set();
+                            BroadcastBlock(peer.Address, BlockChain.Tip);
+                        }
+                        catch (TimeoutException)
+                        {
+                            _logger.Debug($"Timeout occurred during {nameof(ProcessFillblock)}");
+                            await Task.Delay(100);
```

Comment:
How about adding the cancellation token?

```suggestion
                            await Task.Delay(100, cancellationToken);
```